### PR TITLE
chore: Update to operator-rs 0.78.0

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/640

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [x] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -547,29 +552,25 @@ impl SparkApplication {
...
                 let secret_dir = format!("{S3_SECRET_DIR_NAME}/{secret_class_name}");
 
                 // We don't use the credentials at all here but assume they are available
                 submit_cmd.push(format!(
-                    "--conf spark.hadoop.fs.s3a.access.key=$(cat {secret_dir}/{ACCESS_KEY_ID})"
+                    "--conf spark.hadoop.fs.s3a.access.key=\"$(cat {secret_dir}/{ACCESS_KEY_ID})\""
                 ));
                 submit_cmd.push(format!(
-                    "--conf spark.hadoop.fs.s3a.secret.key=$(cat {secret_dir}/{SECRET_ACCESS_KEY})"
+                    "--conf spark.hadoop.fs.s3a.secret.key=\"$(cat {secret_dir}/{SECRET_ACCESS_KEY})\""
```
Comment (NickLarsenNZ) [inline rust/crd/src/lib.rs:573]: Nice :+1:

File: rust/crd/src/s3logdir.rs [hunk 2]
```diff
@@ -120,79 +101,64 @@ impl S3LogDir {
...
+        Ok(result)
     }
 
     fn url(&self) -> String {
-        format!(
-            "s3a://{}/{}",
-            self.bucket.bucket_name.as_ref().unwrap().clone(), // this is guaranteed to exist at this point
-            self.prefix
-        )
+        format!("s3a://{}/{}", self.bucket.bucket_name, self.prefix)
```
Comment (NickLarsenNZ) [inline rust/crd/src/s3logdir.rs:161]:
```suggestion
        format!(
            "s3a://{bucket_name}/{prefix}",
            bucket_name = self.bucket.bucket_name,
            prefix = self.prefix
        );
```

**Review by NickLarsenNZ — DISMISSED** [review]
LGTM

File: rust/crd/src/s3logdir.rs (hunk 2 above)
Comment (sbernauer) [reply rust/crd/src/s3logdir.rs:161]: Thx :)

**Comment by sbernauer** [comment]
Started https://testing.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/spark-k8s-operator-it-custom/7/

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

**Pull request merged**